Kitodo/src/main/java/org/kitodo/services/data/WorkpieceService.java

```diff
@@ -73,19 +75,40 @@ public void saveToIndex(Workpiece workpiece) throws CustomResponseException, IOE
     }
 
     /**
-     * Method saves process and properties related to modified workpiece.
+     * Method manages process and properties related to modified workpiece.
      *
      * @param workpiece
      *            object
      */
     protected void manageDependenciesForIndex(Workpiece workpiece)
             throws CustomResponseException, DataException, IOException {
-        if (workpiece.getProcess() != null) {
-            serviceManager.getProcessService().saveToIndex(workpiece.getProcess());
+        manageProcessDependenciesForIndex(workpiece);
+        managePropertiesDependenciesForIndex(workpiece);
+    }
+
+    private void manageProcessDependenciesForIndex(Workpiece workpiece) throws CustomResponseException, IOException {
+        Process process = workpiece.getProcess();
+        if (workpiece.getIndexAction() == IndexAction.DELETE) {
+            if (process != null) {
```

Comment:
A workpiece should never exists without a corresponding process. It could be possible that a process has no workpiece.